extensions/amp-social-share/0.2/social-share.js

```diff
@@ -30,63 +32,196 @@ const DEFAULT_HEIGHT = 44;
  * @param {!JsonObject} props
  * @return {PreactDef.Renderable}
  */
-export function SocialShare(props) {
+export function SocialShare2(props) {
+  const name = 'SocialShare2';
+  useResourcesNotify();
+
+  const {
+    typeConfig,
+    baseEndpoint,
+    checkedWidth,
+    checkedHeight,
+    obsoleteType,
+  } = checkProps_(props, name);
+  if (obsoleteType) {
+    return;
+  }
+  const finalEndpoint = createEndpoint_(typeConfig, baseEndpoint, props);
+
   /**
-   * Handle key presses on the element.
-   * @param {!Event} event
    * @private
+   * @param {!JsonObject} props
+   * @param {?string} name
+   * @return {!JsonObject}
    */
-  function handleKeyPress(event) {
-    const {key} = event;
-    if (key == Keys.SPACE || key == Keys.ENTER) {
-      event.preventDefault();
-      handleActivation();
+  function checkProps_(props, name) {
+    const {type, shareEndpoint, params, bindings, width, height} = props;
+
+    // Verify type is valid and cannot contain spaces
+    if (type === undefined) {
+      throwError_(`The type attribute is required. ${name}`);
+    }
+    if (/\s/.test(type)) {
+      throwError_(
```

Comment:
This error seems way to detailed and too obscure. Is there a specific reason why we should worry about spaces here?

Reply:
This is a carry over from the previous version of social-share.  I wanted to keep the behavior the same.  I can remove if desired.

Reply:
Let's remove.

Reply:
Removed!